lib/crowbar/client/command/backup/download.rb

```diff
@@ -57,7 +57,7 @@ def write(body)
 
           def path
             filename = Request::Backup::List.new.process do |p|
-              p.parsed_response.select { |s| s["id"] == args.id.to_i }
+              p.parsed_response.select { |s| s["id"] == args.id.to_i || s["name"] == args.id }
```

Comment:
Please use `detect` instead of `select`, that directly returns the first matching row.